### PR TITLE
unsubscribe all users with the same email address

### LINKED ISSUE #12352: Unsubscribing to emails should unsubscribe duplicate accounts as well
#### Issue Description
Campers will often accidentally create duplicate accounts (most of this is because we made the mistake of offering multiple authentication options). 

When they unsubscribe from emails, we want to make sure they are unsubscribed from all emails for all accounts with that email address, including duplicate accounts. 

The unsubscribe code just needs to be refactored to get all accounts with that email address. Currently it's a `findOne` query.

It's located in the RandomAPIs.js and the relevant function is called 

To test your code, you'll want to create multiple accounts that have the same email address. You'll then want to test the unsubscribe functionality (`https://www.freecodecamp.com/unsubscribe/[email]`) and verify that both accounts are unsubscribed. This means all three of these attributes should be set to false:
"sendMonthlyEmail"
"sendNotificationEmail"
"sendQuincyEmail"

### PR DESCRIPTION
**Pull request by suhussai**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #12352 

#### Description

Changed the function call from `findOne()` to `update()` with the `multi` parameter set to `true`. This approach no longer requires the extra step of updating the attributes of the user that was found using the `findOne()` function call. Rather, in one function call, we can update the appropriate attributes of all users that have the same email address. Also, because the update function returns a counter that indicates the number of users that were found, the "Email address not found." error can still be displayed when the counter is equal to zero.

### CODE REVIEW

File: server/boot/randomAPIs.js [hunk 1]
```diff
@@ -171,26 +171,27 @@ module.exports = function(app) {
...
     req.checkParams('email', 'Must send a valid email').isEmail();
-    return User.findOne({ where: { email: req.params.email } }, (err, user) => {
+    var query = { email: req.params.email };
+    var params = {
+      sendQuincyEmail: false,
+      sendMonthlyEmail: false,
+      sendNotificationEmail: false
+    };
+    var options = { multi: true };
+    return User.update(query, params, options, (err, updates) => {
```
Comment (BerkeleyTrue) [inline server/boot/randomAPIs.js]: Did you mean `updateAll`?

http://apidocs.strongloop.com/loopback/v/2.36.2/#persistedmodel-updateall

Also, please change `updates` to `info` so it falls in line with LB docs

File: server/boot/randomAPIs.js [hunk 2]
```diff
@@ -171,26 +171,27 @@ module.exports = function(app) {
...
       if (err) { return next(err); }
-      if (!user) {
+      if (updates['count'] === 0) {
```
Comment (BerkeleyTrue) [inline server/boot/randomAPIs.js]: use dot notation instead of reflection.

`if (updates.count === 0) {`

**Review by BerkeleyTrue — CHANGES_REQUESTED** [review]

File: server/boot/randomAPIs.js (hunk 1 above)
Comment (suhussai) [inline server/boot/randomAPIs.js]: For the `update()` function, I found the documentation in this link: https://mongodb.github.io/node-mongodb-native/api-generated/collection.html. I've never worked with loopback, but it seems to be a middleware between the actual backend and the frontend. Is that correct?

File: server/boot/randomAPIs.js (hunk 2 above)
Comment (suhussai) [inline server/boot/randomAPIs.js]: sure, I can change that.

File: server/boot/randomAPIs.js (hunk 1 above)
Comment (BerkeleyTrue) [inline server/boot/randomAPIs.js]: Yes.

Use the LoopBack api, not the MongoDB api.

Comment (suhussai) [inline server/boot/randomAPIs.js]: I've updated the function call to use updateAll():
```
  function unsubscribeAll(req, res, next) {
    req.checkParams('email', 'Must send a valid email').isEmail();
    var query = { email: req.params.email };
    var params = {
      sendQuincyEmail: false,
      sendMonthlyEmail: false,
      sendNotificationEmail: false
    };
    return User.updateAll(query, params, function(err, info) {
      console.log(err);
      if (err) { return next(err); }
      console.log(info);
      if (info.count === 0) {
        req.flash('info', {
          msg: 'Email address not found. ' +
          'Please update your Email preferences from your profile.'
        });
        return res.redirect('/map');
      } else {
        req.flash('info', {
          msg: 'We\'ve successfully updated your Email preferences.'
        });
        return res.redirect('/unsubscribed');
      }
    });

```

The issue is that `info.count` returns a number larger than 0, but the function call does not update the mongo database. The attributes:

```
"sendMonthlyEmail" : true, 
"sendNotificationEmail" : true,
"sendQuincyEmail" : true
```
remain `true` even after the function call has been made. The weird thing is that the error value is `null` and `info.count` is greater than 0 so the application assumes the change has been made, but when I query the users using the mongo client I can see that it hasn't changed. There are no obvious errors in the console, so I am not sure what's wrong. Any ideas?

Comment (suhussai) [inline server/boot/randomAPIs.js]: I got it now. I wasn't using the `freecodecamp` db in mongodb when I was checking for whether or not the user was being updated. It works now.

**Comment by suhussai** [comment]
@BerkeleyTrue updated the PR.

**Comment by tmpvnunes** [comment]
@suhussai Hi, have you fix this issue?

**Comment by suhussai** [comment]
@tmpvnunes I've tested the new unsubscribe feature and found that it successfully unsubscribes multiple users with the same email.

**Review by BerkeleyTrue — APPROVED** [review]

**Comment by BerkeleyTrue** [comment]
Thanks!

**Pull request merged**